modules/pattern/pattern.resolver.js

```diff
@@ -40,7 +57,22 @@ const patternMutation = {
 
   updatePattern: async (parent, args) => {
     try {
-      return await patternService.updatePattern(args.id, args.pattern);
+      if (!args.upload) {
```

Comment:
I think it is better to move all of this business logic to the `pattern.service.js` layer, let's make resolvers as thin as possible